CMakeLists.txt

```diff
@@ -17,6 +17,7 @@ message("Libvhost log verbosity: ${LIBVHOST_LOG_VERBOSITY}")
 add_library(vhost-server)
 add_compile_definitions(_GNU_SOURCE LOG_VERBOSITY=${LIBVHOST_LOG_VERBOSITY})
 target_compile_options(vhost-server PRIVATE
+  -std=gnu17
```

Comment:
This CMakeLists.txt was introduced by NBS since they integrate libvhost as part of their project, we don't actually build with it locally.
This will need to be checked in with them (might break their build).

Reply:
this code is compiled exclusively by gnu extensions of the standard, it is worth explicitly denoting this

Reply:
@aikuchin Would you be okay with this change?

Reply:
Do we really need 17 here? I was quite sure that 11 was enough, maybe 14, but 17 seems like an overkill and will limit supported compilers significantly.

Reply:
C14 does not exist. C17 is supported by GCC 8.1 and Clang 7. C23 causes confusion.

Reply:
I'd like to understand the motivation for gnu17 if we only use features for gnu11. Why do we need 17, do we ?
If we need new compiler features it is easy to bump requirements in new version, but downgrading standard is always more complicated.

And should we also add required standard version to meson.build?